core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -266,6 +266,11 @@ object SparkSubmit {
       }
     }
 
+    // SPARK-5966, check deployMode CLUSTER and master local
+    if (clusterManager == LOCAL && deployMode == CLUSTER) {
+      printErrorAndExit("Cluster deploy mode is not compatible with master \"local\"")
+    }
```

Comment:
there's a place down there (L321) where we do checks like this. Please move it down there to be more consistent